Source/ChocolateyGui/Services/SettingsSourceService.cs

```diff
@@ -63,13 +54,7 @@ public void RemoveSource(SourceViewModel viewModel)
             }
 
             Settings.Default.sources.Remove(string.Format(CultureInfo.CurrentCulture, "{0}|{1}", viewModel.Name, viewModel.Url));
-            var sourcesChangedEvent = this.SourcesChanged;
-            if (sourcesChangedEvent != null)
-            {
-                sourcesChangedEvent(
-                    this,
-                    new SourcesChangedEventArgs(new List<SourceViewModel>(), new List<SourceViewModel> { viewModel }));
-            }
+            this.RaiseSourceRemovedEvent(viewModel);
```

Comment:
Seeing these makes me wonder if `Settings.Default.sources` could be made in to an `ObservableCollection<T>` to get the `Changed` events automagically? not sure how well the Settings stuff takes to having other collection types, but just a thought to get .Net to do some of the lifting?